stripped/colabs/Detekcja_sezonowosci_stripped.py

```diff
@@ -0,0 +1,915 @@
+  ax1.legend(loc='upper left')
+  ax1.set_ylabel(f"Wartość {attribute}")
+  ax1.set_xlim(xmin, xmax)
+  ax1.grid(True)
+
+  ax2 = plt.subplot(gs[1])
+  ax2.plot(df.index, df['12m_MA'], label='12-miesięczna średnia krocząca', color='red')
+  ax2.set_xlabel('Data')
+  ax2.grid(True)
+  ax2.set_xlim(xmin, xmax)
+
+  plt.tight_layout()
+  plt.show()
+
+
+def perform_augumented_dickey_fuller_print_result(df: pd.DataFrame, attribute: str, prt: bool) -> str:
+    """Funkcja do sprawdzania stacjonarności szeregu czasowego. Wykorzystuje ona Augmented Dickey-Fuller unit root test.
+    Źródło: https://machinelearningmastery.com/time-series-data-stationary-python/
+    Parametry:
+    - df (Pandas DataFrame): ramka danych w Pandas zawierająca co najmniej następujace kolumny: atrybut,
+    - attribute (str): nazwa kolumny, dla której będziemy sprawdzać stacjonarność,
+    - prt (bool): zmienna determinująca, czy wypisać statystyke testu i wartości krytyczne.
+    """
+
+    result = adfuller(df[attribute].values, regression='ct')
+
+    if prt:
+      print(f'ADF Statistic: {round(result[0],4)}')
+      print(f'p-value: {round(result[1], 4)}')
+      print('Critical Values:')
+      for key, value in result[4].items():
+          print(f'{key}, {round(value,3)}')
+
+    if (result[1] <= 0.05) & (result[4]['5%'] > result[0]):
+        return("Stationary")
+    else:
+        return("Non-stationary")
+
+
+def fit_linear_model_and_plot_predictions(df: pd.DataFrame, attribute: str, year_split: int) -> None:
+  """
+  Funkcja wykonuje dummy encoding na kolumnie miesiąc i dopasowuje model regresji liniowej dla jednej lokalizacji, a natępnie generuje wykres predykcji modelu.
+  Parametry:
+  - df (Pandas DataFrame): ramka danych w Pandas zawierająca co najmniej następujace kolumny: Year, Month, atrybut,
+  - attribute (str): nazwa kolumny będąca zmienną zależną (dependent variable),
+  - year_split (int): rok rozdzielający zbiór treningowy i testowy.
+  """
+  assert attribute in df.columns, f"The attribute '{attribute}' is not a column in the DataFrame."
+  #Stworzenie nowej ramki danych, która będzie zawierać dummy variables.
+  df_pd = df.copy(deep=True)
+  # Dummy encoding
+  df_pd = pd.get_dummies(df, columns = ['Month'])
+  # Podział na zbiór treningowy i testowy
+  def train_test_split(df: pd.DataFrame, year_split: int) -> None:
+    """
+    Funckja dzieli ramke danych na zbiór treningowy i testowy.
+    - df (Pandas DataFrame): ramka danych w Pandas zawierająca co najmniej następujace kolumny: Year, Month, atrybut,
+    - year_split (int): rok rozdzielający zbiór treningowy i testowy.
+    """
+    train_data =  df.loc[df['Year'] < year_split]
+    test_data = df.loc[df['Year'] >= year_split]
+    return train_data, test_data
+  train_data, test_data = train_test_split(df_pd, year_split)
+  #Trenowanie modelu
+  regression = LinearRegression().fit(train_data.iloc[:,-12:], train_data[attribute])
```

Comment:
to -12 spełnia warunki antywzorca magic-number [https://www.baeldung.com/cs/antipatterns-magic-numbers](https://www.baeldung.com/cs/antipatterns-magic-numbers)
Mam podejrzenie, że -12 ma coś wspólnego z attribute.
Jeśli np. chcesz przekazać train_data bez jakieś konkretnej kolumny polecam
```
df.drop(columns=[])
```

Reply:
Tego nie zmieniałem, ponieważ według mnie jest to lepsze rozwiązanie. Dodałem opis wyjaśniający skąd bierze się magiczne -12.